src/python/WMComponent/AgentStatusWatcher/DrainStatusAPI.py

```diff
@@ -46,20 +46,26 @@ def checkWorkflows(self):
         results = self.dbsUtil.isAllWorkflowCompleted()
         return results
 
-    def checkCondorStates(self):
+    def checkCondorStates(self, totalOnly=False):
         """
         Check idle and running jobs in Condor
+        :param totalOnly: return the absolute number of idle plus running jobs
+        :return: it returns either an integer or a dictionary, depending on totalOnly argument
         """
-        results = {}
-        queries = [["1", "idle"], ["2", "running"]]
+        results = 0 if totalOnly else {}
+        states = ("Running", "Idle")
 
-        for query in queries:
-            jobs = self.condorAPI.getCondorJobs("JobStatus=="+query[0], [])
+        jobs = self.condorAPI.getCondorJobs()
+        for state in states:
+            if jobs is None and totalOnly:
+                return None
             # if there is an error, report it instead of the length of an empty list
-            if jobs is None:
-                results[query[1]] = "unknown (schedd query error)"
+            elif jobs is None:
+                results[state.lower()] = "unknown (schedd query error)"
+            elif totalOnly:
+                results += int(jobs[0].get(state))
```

Comment:
And actually this should be exactly the way we should go. Because this will also be a softer query to the schedd, since it does not try to return the full list of classads. Here could be one example [1] how this is usually used in  the command line options and in the python bindings respectively. And calling the current function with `TotalOnly=True` is again a full new unique query, which in no way benefits from the fact that you unify the list  of states we query for in a single call.  So we gain nothing by keeping it under the same function, instead of confusion. 

[1]
https://htcondor.readthedocs.io/en/latest/man-pages/condor_q.html
Looking at option `-total`.  And what we should try with python bindings is giving that option as a parameter to the `htcondor.schedd.query()` method. like:

https://htcondor.readthedocs.io/en/latest/apis/python-bindings/api/htcondor.html#htcondor.Schedd.query
https://htcondor.readthedocs.io/en/latest/apis/python-bindings/api/htcondor.html#htcondor.QueryOpts.SummaryOnly

Reply:
Since we are on the topic, I believe the large memory footprint is due to the fact that in the `condorAPI` we use `htcondor.Schedd.query()` instead of `htcondor.schedd.xquery()`. It is mentioned in the warning in the description of the `query()` method in the HTCondor documentation cited above.